quarkus-workshop-super-heroes/docs/src/docs/asciidoc/5a-contract-testing/provider-tests.adoc

```diff
@@ -0,0 +1,160 @@
+== Provider tests
+
+So far, we've done good work and set up tests for the fight service, to extend our testing of the fight service, and also encode its expectations for the hero service into a json contract.
+
+The next step is to actually make sure the contract matches what the hero service is producing!
+
+=== Sharing the contract
+
+There are a few possibilities for sharing the contract.
+A self-hosted or https://pactflow.io/[managed] pact broker can host all of the contracts, as well as providing extra services such as metrics about when tests were run and compatibility dashboards.
+
+Alternatively, some teams check the json contract into the provider's source control after a successful run.
+Note that Pact is consumer-driven contract system; the consumer has responsibility for defining the contract, and therefore also has responsibility for sharing it.
+
+The simplest sharing mechanism is copying on a filesystem, so that's what we'll do.
+From the fights service, copy the contract to the heroes service.
+
+[shell]
+----
+mkdir -p ../rest-heroes/src/test/resources
+cp -r target/pacts ../rest-heroes/src/test/resources
+----
+
+The great thing about the provider Pact tests is that the provider doesn't have to do much.
+The Pact framework will generate a test for each expectation in the contract and run them, based on just a small amount of boilerplate code.
+
+[source,java]
+----
+        @TestTemplate
+        @ExtendWith(PactVerificationInvocationContextProvider.class)
+        void pactVerificationTestTemplate(PactVerificationContext context) {
+            context.verifyInteraction();
+        }
+----
+
+Add the Pact dependency to the heroes service `pom.xml:
+
+[source,xml]
+----
+        <dependency>
+            <groupId>io.quarkiverse.pact</groupId>
+            <artifactId>quarkus-pact-provider</artifactId>
+            <version>{quarkus-pact-version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-panache-mock</artifactId>
+            <scope>test</scope>
+        </dependency>
```

Comment:
You've also added the `quarkus-panache-mock` dependency but you didn't explain why. Before the code block you only mentioned `Add the Pact dependency to the heroes service pom.xml`.